### PR TITLE
Bump CQL Translator/Evaluator to 2.4

### PR DESCRIPTION
**Pull request by vitorpamplona**
**Description**
1. Moves the CQL Evaluator from 2.1 to 2.4 and the translator from 2.1 to 2.4. 
2. Adjusts test cases to match the changes in the CQL-Translator
3. Removes the old `antlr` fix due to the fix being applied in this version of the Translator. 
4. Forces Jackson 2.14.1 due to this bug: https://github.com/FasterXML/jackson-databind/issues/3627. This will be fixed in the next version of HAPI/Translator/Evaluator combo, somewhere in January/February. PR: https://github.com/hapifhir/hapi-fhir/pull/4308
5. Adjusts `generateCarePlan` to return IBaseResource to prepare for R5. 
6. Forces the use of FHIR Core 5.6.36 and HAPI 6.0.1 due to https://github.com/hapifhir/org.hl7.fhir.core/issues/1046
7. Forces Caffeine 2.9.1 while we wait for HAPI 6.3. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Code health

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by vitorpamplona** [comment]
:( We are going to need HAPI 6.3 to push this forward (We need that caffeine/guava modules shipped to Maven central)

**Comment by vitorpamplona** [comment]
This is now ready to be reviewed. @jingtang10 @omarismail94 @ktarasenko @pld 

We need this for a more stable plan definition release.

File: benchmark/build.gradle.kts [hunk 1]
```diff
@@ -73,9 +73,32 @@ configurations {
...
     exclude(module = "jcl-over-slf4j")
     exclude(group = "org.apache.httpcomponents")
-    // Remove this after this issue has been fixed:
-    // https://github.com/cqframework/clinical_quality_language/issues/799
-    exclude(module = "antlr4")
+
+    // Removes newer versions of caffeine and manually imports 2.9
+    // Removes newer versions of hapi and keeps on 6.0.1
+    // (newer versions don't work on Android)
+    resolutionStrategy {
```
Comment (jingtang10) [inline benchmark/build.gradle.kts:80]: can you define a common block for this repeated block?

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -205,8 +235,9 @@ object Dependencies {
     const val fhirUcum = "1.0.3"
     const val guava = "28.2-android"
     const val hapiFhir = "6.0.1"
+    const val hapiFhirCore = "5.6.36"
```
Comment (jingtang10) [inline buildSrc/src/main/kotlin/Dependencies.kt:238]: add an explanation here why

**Review by jingtang10 — CHANGES_REQUESTED** [review]
@vitorpamplona thanks for working this out this is great. why do we need hapi 6.3?

fyi @brynrhodes

File: benchmark/build.gradle.kts (hunk 1 above)
Comment (vitorpamplona) [reply benchmark/build.gradle.kts]: Thanks Jing. I just added a block in the Dependencies file. See if this is better.

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**